lib/src/utils/l10n.dart

```diff
@@ -52,3 +53,50 @@ class _LocaleObserver extends WidgetsBindingObserver {
     _didChangeLocales(locales);
   }
 }
+
+/// Takes a localization function (from lib/l10n/l10n.dart) with a single placeholder
+/// and replaces the placeholder with the given widget, surrounded by Text widgets if necessary.
+///
+/// For example:
+///
+/// ```dart
+/// String translationFun(String name) {
+///   return 'Hello, ${name}!';
+/// }
+///
+/// // returns [Text('Hello, '), MyFancyWidget('Magnus'), Text('!')]
+/// replaceL10nPlaceholderWithWidget(translationFun, MyFancyWidget('Magnus'));
+/// ```
+///
+/// @param l10nWithReplacement The localization function corresponding to a localized string with a placeholder.
```

Comment:
Dart doc comment are not like Java's. Instead of `@param` you need to put the triple slash above the function parameter declaration (line 83 in that case).

Reply:
Whoops, I actually checked dart docs (https://dart.dev/effective-dart/documentation#do-use-prose-to-explain-parameters-return-values-and-exceptions), but overlooked that using `@param` is marked as `bad` :smile: 

The guidelines say to explain the parameters in the description of the method (via square brackets), so I'll do that instead